test.js

```diff
@@ -262,6 +262,43 @@ test('unknown cache key', t => {
   })
 })
 
+test('wills table de-duplicate', t => {
+  t.plan(3)
+  db.flushall()
+  const emitter = mqemitterRedis()
+  const instance = persistence()
+  const client = { id: 'willsTest' }
+
+  instance.broker = toBroker('1', emitter)
+
+  const packet = {
+    cmd: 'publish',
+    topic: 'hello',
+    payload: 'willsTest',
+    qos: 1,
+    retain: false,
+    brokerId: instance.broker.id,
+    brokerCounter: 42,
+    messageId: 123
+  }
+
+  instance.putWill(client, packet, err => {
+    instance.putWill(client, packet, err => {
+      const key = `will:${instance.broker.id}:${encodeURIComponent(client.id)}`
+      instance._db.lrem('will', 0, key, function (err, value) {
```

Comment:
why are you manually calling lrem here on db? shouldn't that be done in putWill ?

Reply:
This particular test relates to non duplication of that particular key, so direct access to lrem or other db function is about the only way to get the total inserted count, which should be 1, not multiples.

Reply:
what about using streamWill instead?